core/src/js/components/sort-dropdown.component.ts

```diff
@@ -0,0 +1,113 @@
+import {
+	AfterViewInit,
+	ChangeDetectionStrategy,
+	ChangeDetectorRef,
+	Component,
+	ElementRef,
+	EventEmitter,
+	Input,
+	Output,
+	ViewRef,
+} from '@angular/core';
+import { IOption } from 'ng-select';
+
+@Component({
+	selector: 'sort-dropdown',
+	styleUrls: [`../../css/global/scrollbar.scss`, `../../css/component/sort-dropdown.component.scss`],
+	template: `
+		<ng-select
+			*ngIf="_visible"
+			class="filter hero-sort-filter"
+			[options]="options"
+			[ngModel]="_filter"
+			[placeholder]="placeholder"
+			(selected)="select($event)"
+			(opened)="refresh()"
+			(closed)="refresh()"
+			[noFilter]="1"
+		>
+			<ng-template #optionTemplate let-option="option">
+				<span
+					[helpTooltip]="option.label"
+					[ngClass]="{ 'unselectable': option?.unselectable }"
+					[innerHTML]="option?.label"
+				></span>
+				<div class="tooltip" *ngIf="option?.tooltip" [helpTooltip]="option.tooltip">
+					<svg>
+						<use xlink:href="assets/svg/sprite.svg#info" />
+					</svg>
+				</div>
+				<i class="i-30 selected-icon" *ngIf="option.value === _filter">
+					<svg class="svg-icon-fill">
+						<use xlink:href="assets/svg/sprite.svg#selected_dropdown" />
+					</svg>
+				</i>
+			</ng-template>
+		</ng-select>
+	`,
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SortDropdownComponent implements AfterViewInit {
```

Comment:
I think here I would use an abstract, parent component that contains the shared code, and use subclasses to redefine the behavior where appropriate. 
The "filter" and "sort" functions are IMO intrinsically very similar, and I don't see them diverging a lot in the future. And if they do, it will most likely be through the options they propose, or how option selection is handled. This means that all the layout is probably best handled in a common way.

Would you like me to do it? Or would you like to give it a try?